### PR TITLE
EC-329 Implement view which allows to specify inputs for transaction 

### PR DESCRIPTION
**Pull request by jhnsmth**

### CODE REVIEW

**Comment by jhnsmth** [comment]
Manually tested

File: app/controllers/ViewController.scala [hunk 1]
```diff
@@ -112,7 +97,7 @@ class ViewController @Inject()(implicit ec: ExecutionContext, ts: TransactionSer
   }
 
   def showSettingsForm(): Action[AnyContent] = Action { implicit request: Request[AnyContent] =>
-    Ok(views.html.settings(settingsForm.fill(ViewController.SettingsData(wallet.flatMap(w => Try(lsmStorage.getWalletSecret(w)).toOption).map(_.privKeyBytes).map(Base58.encode).getOrElse("")))))
+    Ok(views.html.settings(ViewController.settingsForm.fill(ViewController.SettingsData(wallet.flatMap(w => Try(lsmStorage.getWalletSecret(w)).toOption).map(_.privKeyBytes).map(Base16.encode).getOrElse("")))))
```
Comment (GRYE) [inline app/controllers/ViewController.scala]: Try to make this line not sooooo long

File: app/controllers/ViewController.scala [hunk 2]
```diff
@@ -128,11 +113,44 @@ class ViewController @Inject()(implicit ec: ExecutionContext, ts: TransactionSer
...
+    )(ViewController.PaymentTransactionRequestData.apply)(ViewController.PaymentTransactionRequestData.unapply))
+
+  val scriptedTransactionRequestForm: Form[ViewController.ScriptedTransactionRequestData] = Form(
+    mapping(
+      "scriptedTransactionRequest" -> mapping(
+        "fee" -> longNumber,
+        "amount" -> longNumber,
+        "source" -> text
+      )(ScriptedTransactionRequest.apply)(ScriptedTransactionRequest.unapply),
+      "inputsIds" -> text
```
Comment (GRYE) [inline app/controllers/ViewController.scala]: can't we add verification for `inputsIds`?

**Comment by oskin1** [comment]
Transaction creation interface should also allow to specify contract for each specified input.

File: app/controllers/ViewController.scala [hunk 3]
```diff
@@ -21,68 +20,85 @@ class ViewController @Inject()(implicit ec: ExecutionContext, ts: TransactionSer
...
-        str => ts.sendScriptedTransaction(walletId, str).map {
-          _.status match {
-            case StatusCodes.OK => Redirect(routes.ViewController.message("The transaction has been successfully sent to blockchain"))
-            case errorCode => Redirect(routes.ViewController.message("Sending trasaction to the node has failed with code: " + errorCode))
-          }
-        }.recover {
-          case e: IllegalArgumentException => Redirect(routes.ViewController.message(e.getMessage))
-          case e: ExecutionException => Redirect(routes.ViewController.message(e.getCause.getMessage + "\n" + e.getCause.getStackTrace.mkString("\n")))
-          case NonFatal(e) => Redirect(routes.ViewController.message(e.getMessage + "\n" + e.getStackTrace.mkString("\n")))
+        strd => {
```
Comment (ugulavaGeorge) [inline app/controllers/ViewController.scala]: От строк 34-59 отличается вызовом одного метода. Может вынести в функцию?

Comment (jhnsmth) [reply app/controllers/ViewController.scala]: done

File: app/models/EncryAddress.scala [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+object EncryAddress {
+  case object InvalidAddressException extends Exception("Invalid address")
+
+  def resolveAddress(address: String): Try[EncryAddress] = Base58Check.decode(address).map {
+    case bytes if bytes.head == Pay2PubKeyAddress.TypePrefix => Pay2PubKeyAddress(address)
+    case bytes if bytes.head == Pay2ContractHashAddress.TypePrefix => Pay2ContractHashAddress(address)
+  }
+}
+
+/** P2PK - public key */
```
Comment (GusevTimofey) [inline app/models/EncryAddress.scala]: do you need this comment?

File: app/models/EncryAddress.scala [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  def pubKey: PublicKey = decoded.map(PublicKey @@ _.tail).getOrElse(throw EncryAddress.InvalidAddressException)
+  def p2ch: Pay2ContractHashAddress = Pay2ContractHashAddress(PubKeyLockedContract(pubKey))
+}
+object Pay2PubKeyAddress {
+  val TypePrefix: Byte = 0x02
+  def apply(publicKey: PublicKey): Pay2PubKeyAddress = new Pay2PubKeyAddress(Base58Check.encode(TypePrefix +: publicKey))
+  def extractPubKey(address: String): Try[PublicKey] = Base58Check.decode(address).map(PublicKey @@ _.tail)
+}
+
+/** P2CH - regular contract hash */
```
Comment (GusevTimofey) [inline app/models/EncryAddress.scala]: and this

**Pull request merged**